### PR TITLE
Gray out token registration functionality in the WebUI

### PR DESCRIPTION
**Pull request by kelsos**
WebUI Part of #2393

### CODE REVIEW

File: raiden/ui/web/src/assets/config/config.development.json [hunk 1]
```diff
@@ -2,5 +2,6 @@
   "raiden": "/api/1",
   "reveal_timeout": 10,
   "settle_timeout": 500,
-  "web3": "/web3"
+  "web3": "/web3",
+  "network_type": 2
```
Comment (andrevmatos) [inline raiden/ui/web/src/assets/config/config.development.json]: As talked with @LefterisJP , `network_type` should be a lowercase string, either `main` or `test`

File: raiden/ui/web/src/assets/config/config.production.json [hunk 2]
```diff
@@ -2,5 +2,6 @@
   "raiden": "/api/1",
   "reveal_timeout": 10,
   "settle_timeout": 500,
-  "web3": "http://localhost:8545"
+  "web3": "http://localhost:8545",
+  "network_type": 2
```
Comment (andrevmatos) [inline raiden/ui/web/src/assets/config/config.production.json]: Same

File: raiden/ui/web/src/app/services/raiden.config.ts [hunk 3]
```diff
@@ -14,6 +15,7 @@ interface RDNConfig {
     http_timeout?: number;
     settle_timeout?: number;
     reveal_timeout?: number;
+    network_type: number;
```
Comment (andrevmatos) [inline raiden/ui/web/src/app/services/raiden.config.ts]: Please, make it optional and a string: `network_type?: string`, defaulting below to test

File: raiden/ui/web/src/app/services/network-type.enum.ts [hunk 4]
```diff
@@ -0,0 +1,4 @@
+export const enum NetworkType {
+    MAIN = 1,
```
Comment (andrevmatos) [inline raiden/ui/web/src/app/services/network-type.enum.ts]: Don't get tied to the py implementation here.. as you should receive `string` in `config.json`, make the [enum string-based](https://www.typescriptlang.org/docs/handbook/enums.html) according to the lowercase names

File: raiden/ui/web/src/app/components/token-network/token-network.component.ts [hunk 5]
```diff
@@ -47,6 +48,10 @@ export class TokenNetworkComponent implements OnInit, OnDestroy {
 
     public refreshing = true;
 
+    public get main(): boolean {
```
Comment (andrevmatos) [inline raiden/ui/web/src/app/components/token-network/token-network.component.ts]: This getter should be in `RaidenService`, and gotten from there, service is the one that knows better the config, no need to make other components tied to these internal config details

File: raiden/ui/web/src/app/components/token-network/token-network.component.html [hunk 6]
```diff
@@ -14,6 +14,7 @@
             <div fxLayout="row"
                  fxLayoutAlign="end" fxFlex="50">
                 <button mat-icon-button color="accent" (click)="showRegisterDialog()"
+                        [disabled]="main"
```
Comment (andrevmatos) [inline raiden/ui/web/src/app/components/token-network/token-network.component.html]: You may proxy the getter it in the component though, to access from the template

Comment (andrevmatos) [reply raiden/ui/web/src/app/components/token-network/token-network.component.html]: But, anyway, don't simply disable the button, people will start asking. Hide or remove it from the DOM (with *if)

File: raiden/ui/web/src/app/services/raiden.config.ts (hunk 3 above)
Comment (andrevmatos) [reply raiden/ui/web/src/app/services/raiden.config.ts]: Or better, if you change the enum to be a string enum, you can use the enum type directly in the interface I think

File: raiden/ui/web/src/assets/config/config.development.json (hunk 1 above)
Comment (kelsos) [reply raiden/ui/web/src/assets/config/config.development.json]: I will update the PR shortly with the changes :)

**Pull request merged**